### DISCUSSION TITLE
Image size is 10 times after making a small change

### DISCUSSION THREAD

**Discussion by AdeelShadmanQureshi** [dlemstra/Magick.NET#881] [Help]
I have been asked to draw a rectangle over tiff/pdf/png/jpg file of a given size at a given coordinates.

I am using ImageMagick (Magick.NET-Q16-AnyCPU ver 7.23.2.1) for it as it supports TIFF out of the box and using GhostScript it can support PDF.

Using the following code I can add rectangles to any page at a given coordinates.

       
        `using var sourceMs = new MemoryStream(request.Bytes) { Position = 0 };
        using var images = new MagickImageCollection();
        using var ms = new MemoryStream();
        sourceMs.Position = 0;
        images.Read(sourceMs, new MagickReadSettings());
        foreach (var page in request.Pages)
        {
            foreach (var redaction in page.Redactions)
            {
                images[page.PageNumber].Draw(
                    new Drawables()
                        .FillColor(MagickColors.DarkGray)
                        .Rectangle(
                            redaction.XCoordinate,
                            redaction.YCoordinate,
                                redaction.Width,
                                    redaction.Height)
                );
            }
        }

        // I tried this first for reducing file size
        //foreach (var image in images)
        //{
        //    image.SetCompression(CompressionMethod.LosslessJPEG);
        //    image.Scale(new Percentage(50));
        //}

        images.Write(ms, GetFormat(request.MediaType));

        // new trick to reduce size but tiff is not supported.
        ms.Position = 0;
        var optimizer = new ImageOptimizer();
        optimizer.LosslessCompress(ms);

        ms.Position = 0;
        return ms.ToArray();`

However once the rectangle is added - the final file size is 10 times larger than the original file which was fed to the above function.

I tried few things to reduce size by adding compression, scaling and optimiser but nothing worked. I am sure I must be missing something.

Thanks

Note: I posted this question on stackoverflow and someone suggested about bit depth. I checked using the Depth property and it stays at 16 before and after the image manipulation.

**Comment by dlemstra** [comment]
Can you please share an input and output image that demonstrates the problem?

**Reply by AdeelShadmanQureshi** [reply]
[example of size exploding.zip](https://github.com/dlemstra/Magick.NET/files/6176573/example.of.size.exploding.zip)

**Reply by AdeelShadmanQureshi** [reply]
Hi @dlemstra  - Please find attached an example of source and target files. I had a 10 page Tiff which I converted to PDF after adding a rectangle. Let me know if you need more info. Thanks

**Reply by dlemstra** [reply]
I get a 1.4mb file when I execute the following code:

```C#
using (var images = new MagickImageCollection(@"source.tif"))
{
    foreach (var image in images)
    {
        image.Draw(
            new Drawables()
                .FillColor(MagickColors.DarkGray)
                .Rectangle(100, 100, 800, 800));
    }

    images.Write(@"z.pdf");
}
```

Not sure what you are doing differently.

**Reply by AdeelShadmanQureshi** [reply]
thats interesting. The original source image is 4mb. Let me check again.

**Reply by AdeelShadmanQureshi** [reply]
[example 2 of size exploding.zip](https://github.com/dlemstra/Magick.NET/files/6176596/example.2.of.size.exploding.zip)

**Reply by AdeelShadmanQureshi** [reply]
Hi @dlemstra - I took your code and ran it. This time I used a 400kb pdf file and it resulted in 14mb pdf file. I have attached the source and z.pdf plus screenshot of the code. I am using stream as a input. I am reading the file from our data store (blob) - does that matter?

**Reply by dlemstra** [reply]
When converting a PDF file you will convert text to image and that means you will see a big increase in size. There is nothing you can do to prevent that from happening.

**Reply by AdeelShadmanQureshi** [reply]
Right; so Tif to Pdf is the only one combination which reduces size from 4mb to 1.4mb as you experienced it. 4mb tif to tif results in 66mb file size and 400kb pdf results in 14mb. So for now - I am going to go with tif to pdf otherwise, we will never be able to display the file on the UI due to size. Thanks for helping @dlemstra
